lecture1_scala_101/src/main/scala/Lecture.scala

```diff
@@ -0,0 +1,1425 @@
+      <.br,
+      <.h4("Let's do something new. Let's create some Data Types")
+    )
+  )
+
+  val classes = chapter(
+    chapterSlide(
+      <.h2("Classes")
+    ),
+
+    slide(
+      "Create a new type",
+      <.h3("Scala's objects bring us just so far."),
+      <.h4("We want to be able to create our own.")
+    ),
+
+    slide(
+      "Create a new type",
+      <.p("You create new types in Scala by defining Classes. Here, a class is a way to define a set of objects which have same fields and functions."),
+      <.br,
+      Enumeration(
+        Item.fadeIn(<.p("class")),
+        Item.fadeIn(<.p("case class")),
+        Item.fadeIn(<.p("object")),
+        Item.fadeIn(<.p("trait")),
+        Item.fadeIn(<.p("abstract Class"))
+      )
+    ),
+
+    slide(
+      "Classes",
+      code("""
+        class Person(val name: String, val age: Int)
+        //     ^        ^                 ^
+        //     '        '-----------------'
+        // identifier            '
+        //              constructor with public fields
+      """),
+      codeFragment("""
+        val user = new Person("Gandalf", 2019)
+        //          ^
+        //          '
+        //   call constructor / create new object
+
+        user.name == "Gandalf"
+        user.age  == 2019
+      """)
+    ),
+
+    slide(
+      "Classes: private fields",
+      code("""
+        class Person(val name: String, age: Int)
+        //                            ^
+        //                            '
+        //                       private field
+      """),
+      codeFragment("""
+        val user = new Person("Gandalf", 2019)
+
+        user.name == "Gandalf"
+        user.age // not allowed
+      """)
+    ),
+
+    slide(
+      "Classes: inner fields",
+      code("""
+        class Person(val name: String, val age: Int) {
+          private val ageString = age.toString
+
+          val show = "person: " + name + ", " + ageStr
+        }
+
+        new Person("Gandalf", 2019).show == "person: Gandalf, 2019"
+      """)
+    ),
+
+    slide(
+      "Classes: methods",
+      code("""
+        class Person(val name: String, val age: Int) {
+
+          // functions refering to there object are called methods          
+          def isOlder(other: Person): Boolean = age > other.age
+      """),
+      codeFragment("""
+        // equal to
+          def isOlder(other: Person): Boolean = this.age > other.age
+        }
+      """),
+      codeFragment("""
+        // or
+        def isOlder(user: Person)(other: Person): Boolean = user.age > other.age
+      """),
+      codeFragment("""
+        val gandalf = new Person("Gandalf", 2019)
+
+        gandalf.isOlder(new Person("Frodo", 33)) == isOlder(gandalf)(new Person("Frodo", 33))
+      """)
+    ),
+
+    slide(
+      "Classes: methods",
+      <.p("Methods are functions which have their surrounding object in scope and belong to the class. They also provide a means to get infix notation.")
```

Comment:
Maybe it's worth to very briefly talk about open recursion? Or connect back to this later when recursion is introduced.